sqlmesh/core/context.py

```diff
@@ -525,6 +526,7 @@ def snapshots(self) -> t.Dict[str, Snapshot]:
                 config = self.config_for_model(model)
                 ttl = config.snapshot_ttl
                 project = config.project
+                physical_schema = config.physical_schema_map.get(parse_model_name(model.name)[1])
```

Comment:
Shouldn't we inherit the `physical_schema` from the remote snapshot if one was found in the other branch?

Reply:
So I think you should set `physical_schema` to `config.physical_schema_map.get(parse_model_name(model.name)[1])` regardless of whether the `model.name` is in `remote_snapshots` or not. Which also means you can get rid of `physical_schema = None`.